native/rust/word_counter/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+extern crate libc;
+
+use std::error;
+
+use libc::c_char;
+use std::ffi::CStr;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use voca_rs::count;
+
+// Alias Result<T> to dynamically-determined Result<T, (something)>, so we can wrap up all possible result types.
+type Result<T> = std::result::Result<T, Box<dyn error::Error>>;
+
+#[derive(Serialize, Deserialize)]
+pub struct Delta {
+    ops: Vec<DeltaOps>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct DeltaOps {
+    attributes: Option<Value>,
+    insert: Value,
+}
+
+#[no_mangle]
+pub extern "C" fn count_words(work_content: *const c_char) -> u32 {
```

Comment:
If we wanted to change this signature to be capable of passing back an error message, it would become something like:
```rust
pub extern "C" fn count_words(work_content: *const c_char, result_message: *const c_char, result_length: u32) -> u32 {
...
```